sklearn/feature_extraction/text.py

```diff
@@ -242,6 +242,8 @@ def build_analyzer(self):
     def _validate_vocabulary(self):
         vocabulary = self.vocabulary
         if vocabulary is not None:
+            if isinstance(vocabulary, set):
+                self.vocabulary = list(vocabulary)
```

Comment:
I thought the idea was to change it in the method `fit`, since `fit`, and by extension `_validate_vocabulary` make modifications to certain parameters initialised in `init`? Where else would it be appropriate to make this change?

Also, does it matter if the list is sorted?